elements/pfe-primary-detail/src/pfe-primary-detail.js

```diff
@@ -0,0 +1,177 @@
+  }
+
+  static get meta() {
+    return {
+      title: "Primary detail",
+      description: ""
+    };
+  }
+
+  get templateUrl() {
+    return "pfe-primary-detail.html";
+  }
+
+  get styleUrl() {
+    return "pfe-primary-detail.scss";
+  }
+
+  // static get events() {
+  //   return {
+  //   };
+  // }
+
+  // Declare the type of this component
+  static get PfeType() {
+    return PFElement.PfeTypes.Container;
+  }
+
+  static get properties() {
+    return {};
+  }
+
+  constructor() {
+    super(PfePrimaryDetail, { type: PfePrimaryDetail.PfeType });
+
+    this._handleHideShow = this._handleHideShow.bind(this);
+
+    this._detailsNav = this.shadowRoot.getElementById("details-nav");
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    // If you need to initialize any attributes, do that here
+
+    this._slots = {
+      detailsNav: this.shadowRoot.querySelector('slot[name="details-nav"]').assignedElements(),
+      details: this.shadowRoot.querySelector('slot[name="details"]').assignedElements(),
+      detailsNavHeader: this.shadowRoot.querySelector('slot[name="details-nav--header"]').assignedElements(),
+      detailsNavFooter: this.shadowRoot.querySelector('slot[name="details-nav--footer"]').assignedElements()
+    };
+
+    this._detailsWrapper = this.shadowRoot.getElementById("details-wrapper");
+
+    // Add appropriate markup and behaviors
+    this._scanLightDom();
+    // Set first item as active for initial load
+    this._handleHideShow({ target: this._slots.detailsNav[0] });
+  }
+
+  // @todo Add mutation observer in case any content changes, rerun scanLightDom?
+
+  disconnectedCallback() {
+    for (let index = 0; index < this._slots.detailsNav.length; index++) {
+      this._slots.detailsNav[index].removeEventListener("click", this._handleHideShow);
+    }
+  }
+
+  /**
+   * Create nav functionality and adds additional HTML/attributes to markup
+   */
+  _scanLightDom() {
+    if (this._slots.detailsNav.length !== this._slots.details.length) {
+      console.error(
+        `${this.tag}: The number of item headings does not match the number of item details. Found ${this._slots.detailsNav.length} item headings & ${this._slots.details.length} item details.`
+      );
+      return;
+    }
+
+    // Setup left sidebar navigation
+    this._slots.detailsNav.forEach((toggle, index) => {
+      let attr = toggle.attributes;
+      toggle.dataset.index = index;
+
+      const button = document.createElement("button");
+
+      button.innerText = toggle.textContent;
+      // Copy over attributes from original element
+      [...attr].forEach(toggle => {
+        button.setAttribute(toggle.name, toggle.value);
+      });
+
+      // If the toggle does not have a ID, set a unique ID
+      if (!toggle.hasAttribute("id")) {
+        button.setAttribute(
+          "id",
+          `pfe-detail-toggle-${Math.random()
+            .toString(36)
+            .substr(2, 9)}`
+        );
```

Comment:
```suggestion
        button.id = `pfe-detail-toggle-${this.randomId}`;
```

Reply:
That function returns `pfe-` prefixing the random id, which I thought was odd. Wonder if we could change the utility function to:
`get randomId($prefix = 'pfe') {`

Reply:
Oh great idea!  Do you want to make that update or would you like me to?

Reply:
@castastrophe if you have time that'd be awesome, I'm juggling a lot right now.

Reply:
Happy to add that eventually.  I guess I'd ask why `pfe-randomId` isn't sufficient though.  Does the namespacing for the component matter in the ID?

Reply:
It's technically the same, but I think it's a better developer experience for implementors if the ID is more descriptive than `pfe-`.

Reply:
I can respect that but rather than do this one differently than every other component, I think we discuss updating the random ID tool in the base class.  If a system is in place, I think we use that and change the system if it's not the best experience.

Reply:
I thought that's what we were discussing before? I asked if you could do it and didn't get an answer and now ... ? I'm confused.

So I tried to put in a change to add that param, but it's a getter and so it's not allowed to have params.

Creating an issue to discuss: https://github.com/patternfly/patternfly-elements/issues/1251

This PR still has more work, I'm going to leave it for now as a reminder and I'll update it when it gets closer to finished.

Reply:
I think we should leave this as is for now and work on fixing random IDs in a different PR so we don't clutter this up.

Reply:
I agree however I think we should use the existing function and talk about how we want to improve it in the issue rather than hardcode the same functionality here.

Reply:
I've thought about it a lot and I don't agree. I'm creating an identifier that makes it easier for a person to read the DOM. The code that is duplicated is:
```js
Math.random().toString(36).substr(2, 9)
```

Which isn't war and peace.

I don't think it's necessary to standardize _random_ ID's in the project.

Reply:
Why have a system with standards and tools if we don't follow or use them?  I don't think this is so unique and important a case as to need to do something different than every other component generating random IDs.  And just for clarity here - we use these tools to make it easier for us to fix bugs as they appear.  The original randomID getting wasn't pfe-prefixed at all and some of the IDs being generated weren't valid.  We fixed this one bit of code in the base class and the bug was corrected in all components.  That's why standardization is important on large or small tasks.

Reply:
Let's just leave this as is for right now. I've opened an issue #1255 to tackle the broader issue of random ID generation and the things we need to support them like prefixes and suffixes. I think we revisit this once we tackle #1255.